### ISSUE TITLE
Documentation configuration

### ISSUE THREAD

**Issue by nbrucy** [ramses-organisation/ramses#39]
Here are two suggestions to the documentation's configuration:

- [x] Currently, `latest` in the documentation refers to `stable`, while the latest version is actually `dev`
- [x] It would be nice to have auto build activated on PRs, so one can check change they made on the documentation. See https://docs.readthedocs.io/en/stable/tutorial/index.html#triggering-builds-from-pull-requests https://docs.readthedocs.io/en/stable/tutorial/index.html#triggering-builds-from-pull-requests

**Comment by robelgeda** [comment]
@nbrucy  We setup the RTD early, before the dev version had a lead over sable. I think that change should be easy to make, but in the RTD control panel. 

The auto build sounds good to me though usually people should build the docs locally before committing their changes. If this sounds good, I could add instructions on how to do it locally!

**Comment by robelgeda** [comment]
Ah I just realized the goal is to check if RTD builds the updated docs, sorry about the misunderstanding. In that case the your suggestion above is the easiest way!

**Comment by nbrucy** [comment]
Yep that would indeed prevent merging a PR that breaks the doc :smile:

**Comment by robelgeda** [comment]
I have an idea actually. I can create a PR in a few hours!

**Comment by robelgeda** [comment]
PR #42 
I opened a PR with a test for the build, it goes through every step RTD goes through to build the docs. If the test build does not fail but the actual RTD build fails, the issue is with the RTD server setup/config or the user did not check the content of their documentation. I could add code to upload the built pdf as an artifact but I am sure how it will impact our github action limits (probably wont matter to be honest)

CC @nbrucy

**Comment by robelgeda** [comment]
Ok I have enabled uploading artifacts, the docs can be downloaded as a zipped file (html and pf). Steps:
1. Go to the repo actions using the action tab or at: https://github.com/robelgeda/ramses/actions
2. Click on the test run that corresponds to your PR.
3. Click on the 'doc' or download button/link in the artifact section:
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/5174b758-09f5-4016-a7a9-102b8f890b87">

A zipped files should download! 

The RTD build will be under the `html` folder while the pdf will be under `simplepdf/Ramses.pdf`

The artifact should last 90 days but you should also check out the GItHub policy for accounting purposes (seems chill to me):
https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#artifact-and-log-retention-policy

**Comment by robelgeda** [comment]
Ah sorry about this but I noticed they have added a button for building PRs in RTD so that might be a better way to do this:
It should be under `Admin` -> `Settings` -> `Build pull requests for this project`


<img width="269" alt="Screenshot 2024-11-03 at 10 48 00 PM" src="https://github.com/user-attachments/assets/6af651cb-6e0e-46bc-b2cd-b4e3747cf844">

It looks like  rteyssier is the only one with access to the RTD dashboard and access to the settings.

**Comment by nbrucy** [comment]
That would be even better (and what I was thinking of initially).
However the CI improvement you did blocks PRs that break the docs, so that's very good.

Anyway, indeed it's good if @rteyssier also activate this option.

**Comment by rteyssier** [comment]
I am not sure what I am supposed to do…

**Comment by rteyssier** [comment]
I should click in the box right left to "More information in our docs"?

**Comment by robelgeda** [comment]
Yes please @rteyssier, you also have to press save after checking for it to take effect (I always forget to do that). For reference here are the RTD pages about this:
https://docs.readthedocs.io/en/stable/guides/pull-requests.html

**Comment by robelgeda** [comment]
Looks like it's working!

**Comment by tinecolman** [comment]
This is no longer an issue, since we removed the stable branch.

**Comment by nbrucy** [comment]
It is actually, since the docs still points to the last version of stable.

Hi @rteyssier I think some action from you is needed here. You should select `dev` as the default branch in the Read The Docs configuration panel. Here is how to do it:

1. Log in at https://readthedocs.org with your Github accounts
2. Go into the "Configuration" setting for Ramses

![Image](https://github.com/user-attachments/assets/3b088e30-bbdd-427e-8381-fef30cceeb3a)

3.  Select `dev` as the default branch

![Image](https://github.com/user-attachments/assets/eaaf1b12-9053-451d-a4e5-181439d0e662)

4. Save the settings (green button at the bottom of the page)

**Comment by nbrucy** [comment]
It works! Thanks!
